src/metadata/metadata_handler.cc

```diff
@@ -121,16 +121,17 @@ void MetadataHandler::extractMetaData(const std::shared_ptr<Context>& context, c
     if (contentType == CONTENT_TYPE_AVI) {
         std::string fourcc = getAVIFourCC(dirEnt.path());
         if (!fourcc.empty()) {
-            item->getResource(0)->addOption(RESOURCE_OPTION_FOURCC,
-                fourcc);
+            item->getResource(0)->addOption(RESOURCE_OPTION_FOURCC, fourcc);
         }
     }
 #endif // HAVE_FFMPEG
 
 #ifdef HAVE_FFMPEGTHUMBNAILER
     // Thumbnails for videos
     if (itemCls == UPNP_CLASS_VIDEO_ITEM)
-        FfmpegThumbnailerHandler(context).fillMetadata(item);
+        FfmpegThumbnailerHandler(context, CFG_SERVER_EXTOPTS_FFMPEGTHUMBNAILER_VIDEO_ENABLED).fillMetadata(item);
+    else if (itemCls == UPNP_CLASS_IMAGE_ITEM)
+        FfmpegThumbnailerHandler(context, CFG_SERVER_EXTOPTS_FFMPEGTHUMBNAILER_IMAGE_ENABLED).fillMetadata(item);
```

Comment:
I think we should keep the "enabled/disabled" logic out of the handlers themselves, we should just not call them if they are not enabled, and because bool type arguments suck.

Something like `if (itemCls == UPNP_CLASS_IMAGE_ITEM && CFG_SERVER_EXTOPTS_FFMPEGTHUMBNAILER_IMAGE_ENABLED)`

Reply:
Ah, nevermind, I see you pass the Option to use as a flag.

Reply:
Thaught about this, but `serveContent` is called from too many places to squeeze it in here